test/native/rich-test.perf-native.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import {
+	changeTextOfRichText,
+	fireEvent,
+	measurePerformance,
+	screen,
+} from 'test/helpers';
+
+/**
+ * WordPress dependencies
+ */
+import { RichText } from '@wordpress/block-editor';
+
+describe( 'RichText Performance', () => {
+	it( 'performance is stable', async () => {
+		const scenario = async () => {
+			const richTextInput = screen.getByLabelText( 'Text input. Empty' );
+
+			fireEvent( richTextInput, 'focus' );
+
+			changeTextOfRichText(
+				richTextInput,
+				'<strong>Bold</strong> <em>italic</em> <s>strikethrough</s> text'
+			);
+		};
+
+		await measurePerformance( <RichText onChange={ jest.fn() } />, {
+			scenario,
+		} );
+	} );
+
+	it( 'should call onFocus when the TextInput component gains focus', async () => {
+		const scenario = async () => {
+			const richTextInput = screen.getByLabelText( 'Text input. Empty' );
+
+			fireEvent( richTextInput, 'focus' );
+		};
+
+		await measurePerformance( <RichText onFocus={ jest.fn() } />, {
+			scenario,
+		} );
+	} );
+
+	it( 'should call onBlue when the TextInput component is blurred', async () => {
+		const scenario = async () => {
+			const richTextInput = screen.getByLabelText( 'Text input. Empty' );
+
+			fireEvent( richTextInput, 'focus' );
+		};
+
+		await measurePerformance( <RichText onBlur={ jest.fn() } />, {
+			scenario,
+		} );
+	} );
+
+	it( 'should call onSelectionChange when the selection in the TextInput component changes', async () => {
+		const scenario = async () => {
+			const richTextInput = screen.getByLabelText( 'Text input. Empty' );
+
+			const selection = { start: 2, end: 4 };
+			fireEvent( richTextInput, 'selectionChange', {
+				nativeEvent: { selection },
+			} );
```

Comment:
It appears this test fails as the method signature expected when triggering `selectionChange` is the following: 

https://github.com/WordPress/gutenberg/blob/a0fa40d7df3e28decd44a300b1f4053a2df73d21/packages/rich-text/src/component/index.native.js#L644

We would need to tweak the arguments we pass: 

```diff
diff --git a/test/native/rich-test.perf-native.js b/test/native/rich-test.perf-native.js
index 732b3e4a44..9bf9a982f2 100644
--- a/test/native/rich-test.perf-native.js
+++ b/test/native/rich-test.perf-native.js
@@ -60,8 +60,8 @@ describe( 'RichText Performance', () => {
 			const richTextInput = screen.getByLabelText( 'Text input. Empty' );
 
 			const selection = { start: 2, end: 4 };
-			fireEvent( richTextInput, 'selectionChange', {
-				nativeEvent: { selection },
+			fireEvent( richTextInput, 'selectionChange', 2, 4, 'Hello', {
+				nativeEvent: { selection, text: 'Hello' },
 			} );
 		};

```

Reply:
Thanks for the suggestion. I implemented this change, and it is now producing this error:

```
  ● RichText Performance › should call onSelectionChange when the selection in the TextInput component changes

    TypeError: this.props.onChange is not a function

      345 |             // We don't want to refresh if our goal is just to create a record.
      346 |             if ( refresh ) {
    > 347 |                     this.props.onChange( formattedContent );
          |                                ^
      348 |             }
      349 |     }
      350 |

      at RichText.onChange (../../packages/rich-text/src/component/index.native.js:347:15)
```

I'll work on tracking this down as part of this PR.